mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -834,43 +864,51 @@ object GeneralizedLinearRegressionModel extends MLReadable[GeneralizedLinearRegr
 
 /**
  * :: Experimental ::
- * Summarizing Generalized Linear regression Fits.
+ * Summary of [[GeneralizedLinearRegression]] model and predictions.
  *
  * @param predictions predictions output by the model's `transform` method
- * @param predictionCol field in "predictions" which gives the prediction value of each instance
  * @param model the model that should be summarized
- * @param diagInvAtWA diagonal of matrix (A^T * W * A)^-1 in the last iteration
- * @param numIterations number of iterations
- * @param solver the solver algorithm used for model training
  */
 @Since("2.0.0")
 @Experimental
 class GeneralizedLinearRegressionSummary private[regression] (
     @Since("2.0.0") @transient val predictions: DataFrame,
-    @Since("2.0.0") val predictionCol: String,
-    @Since("2.0.0") val model: GeneralizedLinearRegressionModel,
-    private val diagInvAtWA: Array[Double],
-    @Since("2.0.0") val numIterations: Int,
-    @Since("2.0.0") val solver: String) extends Serializable {
+    model: GeneralizedLinearRegressionModel) extends Serializable {
 
   import GeneralizedLinearRegression._
 
-  private lazy val family = Family.fromName(model.getFamily)
-  private lazy val link = if (model.isDefined(model.getParam("link"))) {
+  // Extract values from model immediately to avoid problems with user reseting model Params.
+  // Coefficients is not a deep copy, but that is acceptable.
+
+  /**
+   * Field in "predictions" which gives the prediction value of each instance
+   */
+  @Since("2.0.0")
+  val predictionCol: String = model.getPredictionCol
+
+  private val weightCol: String = model.getWeightCol
+  private val labelCol: String = model.getLabelCol
+
+  private[regression] val familyName: String = model.getFamily
+  private[regression] val family: Family = Family.fromName(familyName)
+  private[regression] val link: Link = if (model.isDefined(model.link)) {
     Link.fromName(model.getLink)
   } else {
     family.defaultLink
   }
+  private[regression] val fitIntercept: Boolean = model.getFitIntercept
+  private[regression] val intercept: Double = model.intercept
+  private[regression] val coefficients: Vector = model.coefficients
```

Comment:
If the model of ```GeneralizedLinearRegressionSummary``` is immutable, it's not necessary to extract these values in advance.